src/cdk-experimental/selection/selection.ts

```diff
@@ -0,0 +1,224 @@
+   * - Modifier Multiple: Multiple items can be selected using modifier keys such as ctrl or shift.
+   */
+  @Input('cdkSelectionMode') mode: 'single' | 'multiple' = 'single';
+
+  /** When multi select is enabled, require a key modifier to select multiples. */
+  @Input('cdkSelectRequireModifier') requireModifier: boolean = false;
+
+  /** Track by used for selection matching. */
+  @Input('cdkSelectionTrackBy') trackBy: (model: T) => any;
+
+  /** Whether the selections can be cleared after selection. */
+  @Input('cdkSelectionDeselectable')
+  get deselectable(): boolean { return this._deselectable; }
+  set deselectable(val) { this._deselectable = coerceBooleanProperty(val); }
+  private _deselectable: boolean = true;
+
+  /** The max number of selections that can be selected */
+  @Input('cdkSelectionMaxSelections')
+  get maxSelections(): number { return this._maxSelections; }
+  set maxSelections(val) {
+    if (val !== undefined && val !== null) {
+      this._maxSelections = coerceNumberProperty(val);
+    }
+  }
+  private _maxSelections: number;
+
+  /** Whether the control is disabled or not. */
+  @Input('cdkSelectionDisabled')
+  get disabled(): boolean { return this._disabled; }
+  set disabled(val) { this._disabled = coerceBooleanProperty(val); }
+  private _disabled: boolean;
+
+  /** Event fired when a selection/deselection occurs. */
+  selectionChange = new EventEmitter<SelectionChange<T>>();
+
+  /** The model backing of the component. */
+  _selectionModel: SelectionModel<T>;
+
+  /** Set of registered toggle components. */
+  _set: any[] = [];
+
+  /** Previously selected index used for range selection. */
+  _previousSelectedIndex = -1;
+
+  constructor(private _elementRef: ElementRef) {}
+
+  ngOnInit() {
+    this. _selectionModel = new SelectionModel<T>(
+      this.mode === 'multiple',
+      this.selections,
+      true,
+      this.trackBy
+    );
+
+    if (this._selectionModel.change) {
+      this._selectionModel.change.subscribe(e => {
+        this.selections = this._selectionModel.selected;
+        this.selectionChange.emit(e);
+      });
+    }
+  }
+
+  /** Register the control. */
+  register(ctrl) {
+    this._set.push(ctrl);
+  }
+
+  /** Deregister a control from the set. */
+  deregister(ctrl) {
+    const idx = this._set.indexOf(ctrl);
+    if (idx > -1) {
+      this._set.splice(idx, 1);
+    }
+  }
+
+  /** Toggle the selection of a item. */
+  toggle(ctrl: any) {
+    if (this.disabled) {
+      return;
+    }
+
+    if (Array.isArray(ctrl.model)) {
+      if (this.mode === 'multiple' &&
+          (this.maxSelections === undefined ||
+          (ctrl.model.length + this._selectionModel.selected.length) < this.maxSelections)) {
+        this._selectionModel.clear();
+        this._selectionModel.toggle(...ctrl.model);
+      }
+    } else {
+
+      this._setSelections(ctrl);
+    }
+  }
+
+  /** Set the user selection for the element. */
+  setTextSelection(type: string) {
```

Comment:
I don't quite follow why this is here

Reply:
When I do SHIFT + CLICK I want to set the text selection on the entire parent directive area to a specified state.

Reply:
There should be a comment with the explanation